fix-for-issue-7631

nmancus1 opened the pull request:
Issue #7631: add code examples

![image](https://user-images.githubusercontent.com/812984/75552047-4d296800-59ea-11ea-90bc-9dd2fcc78b20.png)

Ouput of default example:
```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
   <module name="WhitespaceAfter"/>
  </module>
</module>

$ cat Test.java 
public class Test {

  public void myTest() {
      if (foo) {              // OK
          //...
      } else if(bar) {        // violation
          //...
      }

      testMethod(foo, bar);   // OK
      testMethod(foo,bar);    // violation

      for (;;){}               // OK
      for(;;){}                // violation, space after 'for' is required
  }

}

$ java -jar /var/tmp/checkstyle-8.29-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /var/tmp/Test.java:6:14: 'if' is not followed by whitespace. [WhitespaceAfter]
[ERROR] /var/tmp/Test.java:11:21: ',' is not followed by whitespace. [WhitespaceAfter]
[ERROR] /var/tmp/Test.java:14:7: 'for' is not followed by whitespace. [WhitespaceAfter]
Audit done.
Checkstyle ends with 3 errors.
```

output of non-default example:
```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="WhitespaceAfter">
      <property name="tokens" value="COMMA, SEMI"/>
    </module>
  </module>
</module>

$ cat Test.java 
public class Test {

  public void myTest() {
      int a; int b;           // OK
      int a;int b;            // violation

      testMethod(foo, bar);   // OK
      testMethod(foo,bar);    // violation

      for(;;) {} // OK
  }

}

$ java -jar /var/tmp/checkstyle-8.29-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /var/tmp/Test.java:5:12: ';' is not followed by whitespace. [WhitespaceAfter]
[ERROR] /var/tmp/Test.java:8:21: ',' is not followed by whitespace. [WhitespaceAfter]
Audit done.
Checkstyle ends with 2 errors.

```

rnveach commented:
@nmancus1 Please see https://github.com/checkstyle/checkstyle/wiki/PR-rules .
Specifically number 1, 3, and 4.

nmancus1 commented:
@rnveach thanks for the comment.  To your points:
1. Fixed.
3. The only failing test resulting from `mvn clean verify` is that I've changed the JavaDoc, which should be expected in this case, right?
4. The only CI issue left, besides the failing JavaDoc test, is that the AppVeyor build has failed.  I am struggling to determine the cause of the AppVeyor failure; is it caused by the travis errors?

I appreciate any help you can provide.  Thanks for giving me the opportunity to contribute!

rnveach commented:
>  is that I've changed the JavaDoc, which should be expected in this case, right?

I need more specific information on the failure to be able to help you, but I have an idea what the issue is. You changed the Javadoc but did not update the site's XDOC file. The XDOC is what gets deployed as HTML to the server. The test that is failing verifies that the 2 are kept in synch. You will need to change the appropriate file in the XDOC folder for the test to pass. Look for the one that contains the check and see what it lines up with the Javadoc.

nmancus1 commented:
@rnveach updating the XDOC did the trick!  Thanks for the help.

romani commented on src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:
```diff
@@ -70,7 +70,23 @@
+ *<pre>
+ * public class MyTest {
+ *  public static void myTest() {
+ *      if (foo) { // OK, 'if' followed by whitespace
```
Remove text after "ok" or " violation" in this and other lines. On exact code there is no "if".

nmancus1 replied:
Fixed.

romani commented on src/xdocs/config_whitespace.xml:
```diff
@@ -2268,6 +2268,27 @@ public long toMicros(long d) { return d / (C1 / C0); }
   &lt;property name=&quot;tokens&quot; value=&quot;COMMA, SEMI&quot;/&gt;
 &lt;/module&gt;
         </source>
+              <p>Example:</p>
```
Please add example of behavior on default config.

Please share CLI output of config and example code to prove that Check is actually behave like shown in your example

nmancus1 replied:
<pre><font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> cat MyTest.java  
public class MyTest {
    public static void myTest() {
        if (foo) { // OK
            //...
        } else if(bar) { // violation
            //...
        }

        testMethod(foo, bar); // OK
        testMethod(foo,bar);  // violation

        for (;;) {} // OK
    }
}<span style="background-color:#EFEFEF"><font color="#000000"><b>%</b></font></span>                                                                                                                                                                                                                                             <font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> checkstyle -c my_checks.xml MyTest.java | grep WhitespaceAfter
[WARN] /home/nick/check_style_test/MyTest.java:5:16: &apos;if&apos; is not followed by whitespace. [<font color="#EF2828"><b>WhitespaceAfter</b></font>]
[WARN] /home/nick/check_style_test/MyTest.java:10:23: &apos;,&apos; is not followed by whitespace. [<font color="#EF2828"><b>WhitespaceAfter</b></font>]
<font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> cat my_checks.xml | grep WhitespaceAfter
	&lt;module name=&quot;<font color="#EF2828"><b>WhitespaceAfter</b></font>&quot;/&gt;</pre>

nmancus1 replied:
<pre><font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> cat MyTest.java                                               
public class MyTest {
    public static void myTest() {
        if (foo) { // OK
            //...
        } else if(bar) { // violation
            //...
        }

        testMethod(foo, bar); // OK
        testMethod(foo,bar);  // violation

        for (;;) {} // OK
    }
}
<font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> cat my_checks.xml | grep WhitespaceAfter
	&lt;module name=&quot;<font color="#EF2828"><b>WhitespaceAfter</b></font>&quot;/&gt;
<font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> checkstyle -c my_checks.xml MyTest.java | grep WhitespaceAfter
[WARN] /home/nick/check_style_test/MyTest.java:5:16: &apos;if&apos; is not followed by whitespace. [<font color="#EF2828"><b>WhitespaceAfter</b></font>]
[WARN] /home/nick/check_style_test/MyTest.java:10:23: &apos;,&apos; is not followed by whitespace. [<font color="#EF2828"><b>WhitespaceAfter</b></font>]</pre>

romani replied:
thanks a lot, keep ding this format of prove for each code sample.

Now please make sure that documentation is looks like:
```
<<default config>>
<<code sample>>
some words explanation for config below
<<config>>
<<code sample>>
some words explanation for config below
<<config>>
<<code sample>>
```

you have code sample NOT under default config. Default config is at 2259 line.

nmancus1 replied:
Ok, I see now.  Everything should be in order.  Thanks for your patience!

nmancus1 replied:
<pre><font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> cat MyTest.java                                             
public class MyTest {
    public static void myTest() {
	int a; int b;		 	// OK
	int a;int b;			// violation
		
        testMethod(foo, bar); 	// OK
        testMethod(foo,bar);  	// violation

        for (;;) {} // OK
    }
}
<font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> cat my_checks.xml | sed -n &apos;/WhitespaceAfter/{p;n;p;n;p}&apos;     
	&lt;module name=&quot;WhitespaceAfter&quot;&gt;
  	&lt;property name=&quot;tokens&quot; value=&quot;COMMA, SEMI&quot;/&gt;
	&lt;/module&gt;
<font color="#89E234"><b>➜  </b></font><font color="#34E2E2"><b>check_style_test</b></font> <font color="#729ECF"><b>git:(</b></font><font color="#EF2828"><b>master</b></font><font color="#729ECF"><b>) </b></font><font color="#FBE84F"><b>✗</b></font> checkstyle -c my_checks.xml MyTest.java | grep WhitespaceAfter
[WARN] /home/nick/check_style_test/MyTest.java:4:14: &apos;;&apos; is not followed by whitespace. [<font color="#EF2828"><b>WhitespaceAfter</b></font>]
[WARN] /home/nick/check_style_test/MyTest.java:7:23: &apos;,&apos; is not followed by whitespace. [<font color="#EF2828"><b>WhitespaceAfter</b></font>]</pre>

nmancus1 replied:
OK. I think I've got everything taken care of here, thanks for being patient!

romani commented on src/xdocs/config_whitespace.xml:
```diff
@@ -2258,6 +2258,26 @@ public long toMicros(long d) { return d / (C1 / C0); }
+        for(;;){}               // OK
+    }
+}
+
```
Please remove extra empty line

nmancus1 replied:
Fixed.

romani commented on src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:
```diff
@@ -62,6 +62,23 @@
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class MyTest {
```
Please remove class definition, the less  unnecessary code in example the better.
Your code is not compilable, so extra code does no good.

nmancus1 replied:
Excellent point. Fixed.

romani commented on src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:
```diff
@@ -62,6 +62,23 @@
+ *      testMethod(foo, bar);   // OK
+ *      testMethod(foo,bar);    // violation
+ *
+ *      for(;;){}               // OK
```
Do you know why this is ok ? If this is big it it is better to not put it in example.

nmancus1 replied:
This is the intended behavior: 

> Checks that a token is followed by whitespace, with the exception that it does not check for whitespace after the semicolon of an empty for iterator. Use Check EmptyForIteratorPad to validate empty for iterators.

So, I thought it best to show that.

romani commented:
This PR can be merged ones CI pass.
